frontend/components/RefineButton.tsx

```diff
@@ -1,39 +1,145 @@
-import React from "react";
-import { Button, Flex, Icon } from "@chakra-ui/react";
+import React, { useContext, useState, useEffect } from "react";
+import {
+  Box,
+  Button,
+  Flex,
+  Icon,
+  useToast,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverArrow,
+  PopoverCloseButton,
+  PopoverBody,
+  Slider,
+  SliderTrack,
+  SliderFilledTrack,
+  SliderThumb,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  NumberIncrementStepper,
+  NumberDecrementStepper,
+  Text,
+} from "@chakra-ui/react";
 import { MdSettings } from "react-icons/md";
+import { WalletContext } from "@/components/wallet/WalletContext";
+import { BurntPixArchives__factory } from "@/contracts";
+import SignInButton from "./SignInButton";
 import { inter } from "@/app/fonts";
 
 const RefineButton: React.FC = () => {
-  const buttonBgColor = "#FE005B";
-  const buttonTextColor = "white";
-  const iconButtonSize = "24px";
+  const walletContext = useContext(WalletContext);
+  const { account, provider, networkConfig } = walletContext;
+  const defaultIterations = 100;
+  const [selectedIterations, setSelectedIterations] =
+    useState(defaultIterations);
+  const toast = useToast();
+  const maxIterations = 10000;
+  const defaultRed = "#FE005B";
+  const burntPixArchives = BurntPixArchives__factory.connect(
+    networkConfig.burntPixArchivesAddress,
+    provider,
+  );
 
-  return (
-    <Flex align="center" justify="right">
-      <Button
-        bg={buttonBgColor}
-        color={buttonTextColor}
-        _hover={{ bg: "red.600" }}
-        borderRadius={10}
-        h="30px"
-        w="70px"
-        fontSize="md"
-        fontFamily={inter.style.fontFamily}
-        fontWeight={700}
-      >
-        REFINE
-      </Button>
+  const refine = async () => {
+    try {
+      console.log("Refining with iterations: ", selectedIterations);
+      const signer = await provider.getSigner();
+      await burntPixArchives
+        .connect(signer)
+        ["refineToArchive(uint256)"](selectedIterations);
+    } catch (error: any) {
+      toast({
+        title: `Error refining. ${error.message}`,
+        status: "error",
+        position: "bottom-left",
+        duration: 5000,
+        isClosable: true,
+      });
```

Comment:
cancelling the transaction popup shows a big message in toast
<img width="644" alt="Screenshot 2024-04-04 at 19 34 32" src="https://github.com/yearone-io/burntpix-archives/assets/132473036/e36d404f-904f-40b0-a29d-db49537a3f9b">